data_structures/src/chain.rs

```diff
@@ -2929,8 +2930,21 @@ fn compare_reputed_pkh(
     let rep_b = rep_eng.trs().get(b).0;
 
     rep_a.cmp(&rep_b).then_with(|| {
-        // FIXME(#1507): implement better tie breaker
-        a.cmp(b)
+        if rep_a > 0 {
+            let alpha_bytes = rep_eng.current_alpha.0.to_be_bytes().to_vec();
+            let mut a_bytes = a.hash.to_vec();
+            let mut b_bytes = b.hash.to_vec();
+
+            a_bytes.extend(alpha_bytes.clone());
+            b_bytes.extend(alpha_bytes);
+
+            let new_hash_a: Hash = calculate_sha256(&a_bytes).into();
+            let new_hash_b: Hash = calculate_sha256(&b_bytes).into();
+
+            new_hash_a.cmp(&new_hash_b)
+        } else {
+            a.cmp(&b)
```

Comment:
Good idea. But explain why:

```suggestion
            // If both identities have 0 reputation their ordering is not important because they will have the same eligibility, so compare them by PublicKeyHash
            a.cmp(&b)
```